src/transmission-utils.ts

```diff
@@ -0,0 +1,54 @@
+import { FinalityType } from "./socket-types";
+
+export const decodeTransmissionParams = (transmissionParam: string) => {
+  if (transmissionParam.length !== 64 && transmissionParam.length !== 66) {
+    throw new Error("Invalid transmission param length");
+  }
+  transmissionParam = transmissionParam.replace("0x", "");
+  const version = parseInt("0x" + transmissionParam.slice(0, 2));
+  const finalityType = parseInt("0x" + transmissionParam.slice(2, 4)); // bucket, block, time
+  const value = parseInt("0x" + transmissionParam.slice(4, 12));
+
+  return { version, finalityType, value };
+};
+
+export const encodeTransmissionParams = (
+  version: number,
+  finalityType: number,
+  value: number
+) => {
+  let transmissionParam = "0x";
+  transmissionParam += version.toString(16).padStart(2, "0");
+  transmissionParam += finalityType.toString(16).padStart(2, "0");
+  transmissionParam += value.toString(16).padStart(8, "0");
+  transmissionParam = transmissionParam.padEnd(66, "0");
+  return transmissionParam;
+};
+
+export const isTxFinalized = (data: {
+  type: FinalityType;
+  currentBlock: number;
+  eventBlock: number;
+  currentTime: number;
+  eventTime: number;
+  finalityBlockDiff: number;
+  finalityTimeDiff: number;
+}) => {
```

Comment:
why is there data object here? not consistent with other functions where we simple take typed params as input